File: 2020_2021/L3_S5_2020-2021/INFO0503_modélisation web serv & web avancé/INFO503/TP2_sofiane/Moteur.php
<?php
include("Carburation.php");
class Moteur{

    const PUISSDEF = 110;
    // attributs

    protected $carburation;
    protected $puissance;
    static $id = 0;


    // constructeur par initialisation
    public function __construct($carburation,$puissance){
    	self::$id++;
        self::$id = self::$id;
        $this->carburation = $carburation;
        if($puissance < 1){
            $this->puissance = PUISSDEF;
        }else{
            $this->puissance = $puissance;
        }
    }

    // getter

    public static function getPuissDef(){
        return self::PUISSDEF;
    }

    public function getPuissance(){
        return $this->puissance;
    }

    public function getCarburation(){
        return $this->carburation;
    }

    public  function getId(){
        return self::$id;
    }

    //setters

    public function setCarburation(Carburation $carburation){
        $this->carburation = $carburation;
    }

    public function setPuissance(int $puissance){
        $this->puissance = $puissance;
    }

    // methode toString

    public function __toString():String{
        return " PuissDef : " . $this->getPuissDef() . " id : " .$this->getId() . " carburation : " . $this->getCarburation() . " puissance : " . $this->getPuissance();
    }
}
?>

File: 2020_2021/L3_S5_2020-2021/INFO0503_modélisation web serv & web avancé/INFO503/TP5_projet_1_Sofiane/Projet_1_MEDDOURI_VANSCOOR/Moteur.java
public class Moteur extends Voiture{
    //Constantes
    static final int PUISSDEF = 110;
  
    //Attributs
    static int id = 0;
  
    protected Carburation carburation;
    protected int puissance;
  
    //Constucteur
    public Moteur(){
    super();
    id++;
    this.id = id;
    carburation = Carburation.DIESEL;
    puissance = PUISSDEF;
    }
  
    public Moteur(Carburation carbu, int puissance){
        id++;
        this.id = id;
        carburation = carbu;
  
        if(puissance<1){
            this.puissance = PUISSDEF;
        }
        else{
            this.puissance = puissance;
        }
    }

    // getter

    public static int getPuissDef(){
        return PUISSDEF;
    }

    public static int getId(){
        return id;
    }

    public Carburation getCarburation(){
        return carburation;
    }

    public int getPuissance(){
        return puissance;
    }

    // setter

    public void setCarburation(Carburation carburation){
        this.carburation = carburation;
    }

    public void setPuissance(int puissance){
        this.puissance = puissance;
    }

    // méthode toString

    public String toString(){
        return " PuissDef : " + getPuissDef() + " id : " + getId() + " carburation : " + getCarburation() + " puissance : " + getPuissance();
    }
}